app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/presentation/common/Presenter.kt

```diff
@@ -0,0 +1,12 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common
+
+import io.reactivex.disposables.CompositeDisposable
+
+abstract class Presenter {
+
+    var subscriptions = CompositeDisposable()
+
+    fun onActivityDestroy() {
+        subscriptions.clear()
```

Comment:
you probably want to call `dispose()` not `clear()` as clear leaves it open for reuse which does not make sense when activity is destroyed.